frame/assets/src/lib.rs

```diff
@@ -343,28 +575,23 @@ pub mod pallet {
 		///
 		/// Emits `Destroyed` event when successful.
 		///
-		/// Weight: `O(1)`
-		#[pallet::weight(T::WeightInfo::force_destroy(*zombies_witness))]
+		/// Weight: `O(c + p + a)` where:
+		/// - `c = (witness.accounts - witness.sufficients)`
+		/// - `s = witness.sufficients`
+		/// - `a = witness.approvals`
+		#[pallet::weight(T::WeightInfo::force_destroy(
+			witness.accounts.saturating_sub(witness.sufficients),
+ 			witness.sufficients,
+ 			witness.approvals,
+ 		))]
 		pub(super) fn force_destroy(
```

Comment:
You could collapse `destroy` and `force_destroy` by first checking if it is the force origin, and if not, then falling back to ensure signed. Imo a cleaner API? and less redundant code.